tests/test_preprocessing/test_interval_encoder.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+import numpy as np
+from sklearn.utils import estimator_checks
+
+from sklego.common import flatten
+from sklego.preprocessing import IntervalEncoder
+from tests.conftest import transformer_checks, general_checks
+
+
+@pytest.mark.parametrize(
+    "test_fn",
+    flatten(
+        [
+            transformer_checks,
+            general_checks,
+            estimator_checks.check_estimators_dtypes,
+            estimator_checks.check_fit_score_takes_y,
+            # estimator_checks.check_dtype_object,
+            estimator_checks.check_sample_weights_pandas_series,
+            estimator_checks.check_sample_weights_list,
+            estimator_checks.check_sample_weights_invariance,
+            estimator_checks.check_estimators_fit_returns_self,
+            estimator_checks.check_complex_data,
+            estimator_checks.check_estimators_empty_data_messages,
+            estimator_checks.check_pipeline_consistency,
+            estimator_checks.check_estimators_nan_inf,
+            estimator_checks.check_estimators_overwrite_params,
+            estimator_checks.check_estimator_sparse_data,
+            estimator_checks.check_estimators_pickle,
+        ]
+    ),
+)
+def test_estimator_checks(test_fn):
+    test_fn(IntervalEncoder.__name__, IntervalEncoder(n_chunks=2))
```

Comment:
Should we also test for `n_chunks=1` to better cover the extremities?

Reply:
It will result in the mean ... but that makes it a good test. Noted!

Reply:
added.